application/build.gradle

```diff
@@ -10,29 +10,37 @@ sourceCompatibility = '17'
 
 repositories {
   mavenCentral()
+  mavenLocal()
 }
 
 ext {
-  set('awsSpringVersion', '2.4.2')
-  set('testcontainersVersion', '1.17.5')
-  set('selenium.version', '4.5.0')
+  set('awsSpringVersion', '3.0.0-RC1')
+  set('testcontainersVersion', '1.17.6')
+  set('thymeleafExtrasSpringSecurityVersion', '3.1.1.RELEASE')
+  set('springBootStarterActiveMQVersion', '2.7.8')
+  set('jsr305Version', '3.0.2')
+  set('logbackAwslogsJsonEncoderVersion', '1.1.1')
+  set('errorHandlingSpringBootStarterVersion', '4.1.0')
+  set('selenideVersion', '6.11.0')
 }
 
 dependencies {
   implementation 'org.springframework.boot:spring-boot-starter-oauth2-client'
   implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'
-  implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity5'
+  implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity6:${thymeleafExtrasSpringSecurityVersion}"
   implementation 'org.springframework.boot:spring-boot-starter-web'
   implementation 'org.springframework.boot:spring-boot-starter-mail'
   implementation 'org.springframework.boot:spring-boot-starter-webflux'
   implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
   implementation 'org.springframework.boot:spring-boot-starter-websocket'
   implementation 'org.springframework.boot:spring-boot-starter-validation'
   implementation 'org.springframework.boot:spring-boot-starter-actuator'
-  implementation 'io.awspring.cloud:spring-cloud-starter-aws'
-  implementation 'io.awspring.cloud:spring-cloud-starter-aws-messaging'
+  implementation "io.awspring.cloud:spring-cloud-aws-core:${awsSpringVersion}"
+  implementation 'io.awspring.cloud:spring-cloud-aws-starter'
+  implementation 'io.awspring.cloud:spring-cloud-aws-starter-ses'
+  implementation 'io.awspring.cloud:spring-cloud-aws-starter-sqs'
   implementation 'io.micrometer:micrometer-registry-cloudwatch'
-  implementation 'org.springframework.boot:spring-boot-starter-activemq'
+  implementation "org.springframework.boot:spring-boot-starter-activemq:${springBootStarterActiveMQVersion}"
```

Comment:
What's the reason for not using the 3.X ActiveMQ starter? Mixing 2.X and 3.X Spring Boot versions may lead to incompatibilities during runtime

Reply:
Unfortunately, there is no ActiveMQ starter for Spring Boot 3 and it seems like there isn't going to be one:

https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-activemq
https://stackoverflow.com/questions/74653414/migrating-to-spring-boot-3-with-activemq-classic

For the moment, this seems to be working but for the future we maybe could come up with a different solution for this use case. However, that kind of refactoring probably is way out of scope for this update as it would involve major changes to the relevant chapter in the ebook as well.